### ISSUE TITLE
Document or make default docker hub URL

### ISSUE THREAD

**Issue by celskeggs** [sipb/homeworld#461]
Currently, there's a prefix one has to use to pull down containers from the docker registry. It's either `docker.io` or `registry.hub.docker.com`, and a `library/` prefix is required (maybe?) if there's no slash in the container name.

If you don't include it, you get the following (via quentin):

> Failed to inspect image "ubuntu": rpc error: code = Unknown desc = no registries configured while trying to pull an unqualified image, add at least one in either /etc/crio/crio.conf or /etc/containers/registries.conf

We should consider whether we want to make the docker registry used by default. From a usability perspective, we do, but from a security perspective, maybe we don't? I'm not sure. If we don't, we'll need to document this.

**Comment by cryslith** [comment]
We could potentially have the best of both worlds by enabling this default but writing a webhook which requires images deployed in non-user namespaces to come from a whitelist of container registries.

Related: https://github.com/kubernetes/kubernetes/issues/22888#issuecomment-488054384

**Comment by celskeggs** [comment]
I don't think the concern is people using weird container registries; we already let users use whatever container registry they want. I think it's more about "do we actually want the docker hub to be the default when someone doesn't specify otherwise"

**Comment by cryslith** [comment]
> requires images deployed in **non-user** namespaces to come from a whitelist of container registries

As in, to mitigate the concern that we might accidentally pull something from docker hub into a privileged namespace.  Not to restrict users.

Anyway, I think unfortunately most users will expect dockerhub to be the default, and won't want this default disabled as a security "feature".

**Comment by celskeggs** [comment]
Yes, missed that -- you're right. I suspect that this kind of restriction is a good idea, but that it will be a later-stage feature, since I don't think we want to put in the work to set up a webhook yet.
